toy_problem/src/main.cpp

```diff
@@ -1,42 +1,130 @@
 #include "TaskAllocation.cpp"
+#include "yaml-cpp/yaml.h"
+#include <iostream>
+#include <fstream>
 
-int main()
+class TaskConfig
```

Comment:
Instead of having the parsing stuff defined in `main()` I recommend having separate source file `parse_tasks.cpp` with a `parse_yaml()` method defined. This method should return a `std::pair<std::vector<ConstTaskRequestPtr>, std::vector<RobotState>>` which can directly be used to initialize the `TaskPlanner` instance. All the parameters for battery, mechanical and power systems can be defined in the yaml file under the `agents` node.

Reply:
For reference https://github.com/osrf/rmf_core/blob/master/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/parse_graph.hpp#L33

Reply:
Split the parsing component a separate file. As mentioned before, will make these proposed changes to a future PR.

Reply:
Thanks for splitting the parsing into a separate file! Why not tackle the other changes in the same PR since its introducing the same feature? But If you strongly prefer, I can merge this in.

Reply:
Sure can. the current grid implementation is for fast sanity check. If using the `nav.yaml` , we will need to draw and then generated it with `traffic_editor`. As the current one doesn't consider about moving along predefined lanes, I suppose this will be trickier to construct on matplotlib visualization, hence need a longer time to implement.